tlm_teamd/values_store.cpp

```diff
@@ -144,6 +144,47 @@ std::string ValuesStore::get_value(json_t * root, const std::string & path, Valu
     throw std::runtime_error("Reach the end of the ValuesStore::get_value. Path=" + path);
 }
 
+///
+/// Extract static lag values for LAG with name lag_name, from the parsed json tree with root, to the temporary storage
+/// @param lag_name a name of the LAG
+/// @param root a pointer to the parsed json tree
+/// @param storage a reference to the temporary storage
+///
+
+void ValuesStore::extract_values_static(const std::string & lag_name, json_t * root, HashOfRecords & storage)
```

Comment:
Why do we need to have a separate function to extract values for static? This appears to be duplicate of the existing API. Can you clarify the difference?

Reply:
teamdctl state information for Static lag and Dynamic LACP are little different.

Dynamic LACP has the following fields along with setup
{
    "ports": {
        "Ethernet48": {
            "ifinfo": {
                "dev_addr": "00:e0:ec:cc:a8:de",
                "dev_addr_len": 6,
                "ifindex": 40,
                "ifname": "Ethernet48"
            },
            "link": {
                "duplex": "half",
                "speed": 0,
                "up": false
            },
            "link_watches": {
                "list": {
                    "link_watch_0": {
                        "delay_down": 0,
                        "delay_up": 0,
                        "down_count": 5,
                        "name": "ethtool",
                        "up": false
                    }
                },
                "up": false
            },
            "runner": {
                "actor_lacpdu_info": {
                    "key": 103,
                    "port": 49,
                    "port_priority": 255,
                    "state": 5,
                    "system": "00:e0:ec:cc:a8:de",
                    "system_priority": 65535
                },
                "aggregator": {
                    "id": 0,
                    "selected": false
                },
                "key": 103,
                "partner_lacpdu_info": {
                    "key": 0,
                    "port": 0,
                    "port_priority": 0,
                    "state": 0,
                    "system": "00:00:00:00:00:00",
                    "system_priority": 0
                },
                "partner_retry_count": 3,
                "prio": 255,
                "selected": false,
                "state": "disabled"
            }
        }
    },
    **"runner": {
        "active": true,
        "enable_retry_count_feature": true,
        "fallback": true,
        "fast_rate": false,
        "retry_count": 3,
        "select_policy": "lacp_prio",
        "sys_prio": 65535
    },
    "setup": {
        "daemonized": true,
        "dbus_enabled": false,
        "debug_level": 1,
        "kernel_team_mode_name": "loadbalance",
        "pid": 297,
        "pid_file": "/var/run/teamd/PortChannel03.pid",
        "runner_name": "lacp",
        "zmq_enabled": false
    },**
    "team_device": {
        "ifinfo": {
            "dev_addr": "00:e0:ec:cc:a8:de",
            "dev_addr_len": 6,
            "ifindex": 50,
            "ifname": "PortChannel03"
        }
    }
}

Static Lag will not contain the runner option (Dynamic LACP needs runner option to determine peer lacp link protocol information)

{
    "ports": {
        "Ethernet32": {
            "ifinfo": {
                "dev_addr": "00:e0:ec:cc:a8:de",
                "dev_addr_len": 6,
                "ifindex": 39,
                "ifname": "Ethernet32"
            },
            "link": {
                "duplex": "half",
                "speed": 0,
                "up": true
            },
            "link_watches": {
                "list": {
                    "link_watch_0": {
                        "delay_down": 0,
                        "delay_up": 0,
                        "down_count": 3,
                        "name": "ethtool",
                        "up": true
                    }
                },
                "up": true
            }
        },
        "Ethernet64": {
            "ifinfo": {
                "dev_addr": "00:e0:ec:cc:a8:de",
                "dev_addr_len": 6,
                "ifindex": 37,
                "ifname": "Ethernet64"
            },
            "link": {
                "duplex": "half",
                "speed": 0,
                "up": false
            },
            "link_watches": {
                "list": {
                    "link_watch_0": {
                        "delay_down": 0,
                        "delay_up": 0,
                        "down_count": 3,
                        "name": "ethtool",
                        "up": false
                    }
                },
                "up": false
            }
        }
    },
    "setup": {
        "daemonized": true,
        "dbus_enabled": false,
        "debug_level": 1,
        "kernel_team_mode_name": "roundrobin",
        "pid": 49,
        "pid_file": "/var/run/teamd/PortChannel02.pid",
        "runner_name": "roundrobin",
        "zmq_enabled": false
    },
    "team_device": {
        "ifinfo": {
            "dev_addr": "00:e0:ec:cc:a8:de",
            "dev_addr_len": 6,
            "ifindex": 9,
            "ifname": "PortChannel02"
        }
    }
}

Since Static doesnt have those runner information modifying existing LAG Member table done for LACP can create more collaterals, thats the reason for separating it out between Static and LACP.